src/Enterspeed.Source.UmbracoCms.Base/Data/Repositories/EnterspeedJobRepository.cs

```diff
@@ -203,13 +204,20 @@ public void Update(EnterspeedJob job)
 
         public void Delete(IList<int> ids)
         {
-            using (var scope = _scopeProvider.CreateScope())
+            // The maximum number of parameters allowed in a single database query is 2000.
+            // This limit is based on typical database constraints (e.g., SQL Server's parameter limit).
+            const int maxParams = 2000;
+            using var scope = _scopeProvider.CreateScope();
+            foreach (var batch in ids.Select((id, idx) => new { id, idx })
```

Comment:
Could this be simplified by using `.Chunk(maxParams)`?

Reply:
Chunk is not available in .NET 5, which would give issues for Umbraco 9

Reply:
Fair point, we can't use `chunk`. Maybe you can create your own `chunk` extension method then, making it easier to read and can be reused.